### DISCUSSION TITLE
Allow Open in new window for buttons, tabs, nav_items?

### DISCUSSION THREAD

**Discussion by DeeJaayArgh** [h2oai/wave#1930] [Q&A] [answered]
As an example, using either tab.py or nav.py in examples folder, I'm wondering if its possible to implement the ability for the user to use a button/tab/nav_item and open the target in a new tab via the ctrl button or browser context menu? 

I see that there are is a link parameter for buttons and tabs, but this affects just the rendering of the component and not its function.

**Comment by mturoci** [comment]
Button has a `path` attr that might be what you need - https://wave.h2o.ai/docs/widgets/form/button/#with-path.

As for tab, I am not sure it makes much sense since tabs usually "toggle" different parts of UI, so everything happens in the same page.

Nav_item could use a `path` attr, please file a feature request.

**Reply by DeeJaayArgh** [reply]
I could see it still being worthwhile for tabs if a user wants to be able to quickly view both tabs at the same time in a dual monitor setup.  If the paths are internal links, developers would just have to think about how much client state would need to be encoded into the hash routing of the path.

As for button, when path is provided and the button is clicked the target appears in a new browser tab and does not replace the current tab, not sure if that was by design. Was it?  Maybe there needs to be an command or internal_path parameter to handle hash routes?

**Reply by mturoci** [reply]
> I could see it still being worthwhile for tabs if a user wants to be able to quickly view both tabs at the same time in a dual monitor setup. If the paths are internal links, developers would just have to think about how much client state would need to be encoded into the hash routing of the path.

That's a fair point.

> As for button, when path is provided and the button is clicked the target appears in a new browser tab and does not replace the current tab, not sure if that was by design. Was it?

Opening in a new window is by design.

>  Maybe there needs to be an command or internal_path parameter to handle hash routes?

I am not sure I follow (maybe due to Friday). That's what `name='#my-hash-route'` is for *supported on both button and tab), isn't it?

**Reply by DeeJaayArgh** [reply]
> I am not sure I follow (maybe due to Friday). That's what name='#my-hash-route' is for *supported on both button and tab), isn't it?

Oh yes, you're right.  I'm farther from the weekend so I've less excuse. 

Here's how button **currently** works:  

- if `name='my-hash-route` 
  - clicking the button results in navigation to `'my-hash-route'` staying within the same browser tab. 
- if the button has `path='my-hash-route'` 
  - clicking the button creates a new browser tab with the app navigated to `'my-hash-route'` (minus any q.client state values).  

I guess what I'm looking for is for the user to have control over how the navigation occurs either: within browser tab, in a new browser tab, or in new browser window.

**Reply by mturoci** [reply]
> I guess what I'm looking for is for the user to have control over how the navigation occurs either: within browser tab, in a new browser tab, or in new browser window.

I see. Let me answer that on Monday.

**Reply by mturoci** [reply, marked as answer]
> I guess what I'm looking for is for the user to have control over how the navigation occurs either: within browser tab, in a new browser tab, or in new browser window.

This makes sense and should be doable within React-router.
